druntime/src/__importc_msvc_builtins.d

```diff
@@ -0,0 +1,12888 @@
+            }
+            else
+            {
+                version (LDC)
+                {
+                    import ldc.llvmasm : __ir_pure;
+
+                    return __ir_pure!(
+                        `%result = call {i64, i64} asm
+                             "div $4",
+                             "={rax},={rdx},0,1,r,~{flags}"
+                             (i64 %1, i64 %0, i64 %2)
+
+                         %quotient = extractvalue {i64, i64} %result, 0
+                         %remainder = extractvalue {i64, i64} %result, 1
+
+                         store i64 %remainder, ` ~ llvmIRPtr!"i64" ~ ` %3
+                         ret i64 %quotient`,
+                        ulong
+                    )(highDividend, lowDividend, divisor, remainder);
+                }
+                else version (GNU)
+                {
+                    ulong quotient;
+                    ulong remainer;
+
+                    asm @trusted pure nothrow @nogc
+                    {
+                          "div %4"
+                        : "=a" (quotient), "=d" (remainer)
+                        : "0" (lowDividend), "1" (highDividend), "rm" (divisor)
+                        : "cc";
+                    }
+
+                    *remainder = remainer;
+                    return quotient;
+                }
+                else version (D_InlineAsm_X86_64)
+                {
+                    asm @trusted pure nothrow @nogc
+                    {
+                        /* RCX is highDividend; RDX is lowDividend; R8 is divisor. R9 is remainder. */
+                        naked;
+                        mov RAX, RDX;
+                        mov RDX, RCX;
+                        div R8;
+                        mov [R9], RDX;
+                        ret;
+                    }
+                }
+            }
+        }
+    }
+
+    version (X86_64_Or_X86)
+    {
+        extern(C)
+        pragma(inline, true)
+        int _div64(long dividend, int divisor, scope int* remainder) @safe pure nothrow @nogc
+        {
+            if (__ctfe)
+            {
+                import core.internal.string : signedToTempString;
+
+                if (((dividend < 0 ? -dividend : dividend) >>> 32) >= (divisor < 0 ? -divisor : divisor))
+                {
+                    /* The div instruction will raise a #DE exception on overflow or division-by-zero,
+                       so during CTFE we'll just assert false. */
+                    assert(
+                        false,
+                          "Division by zero, or an overflow of the 32-bit quotient occurred in _div64."
+                        ~ " dividend: " ~ signedToTempString(dividend)
+                        ~ "; divisor: " ~ signedToTempString(divisor)
+                    );
+                }
+
+                *remainder = cast(int) (dividend % divisor);
+                return cast(int) (dividend / divisor);
+            }
+            else
```

Comment:
Is it really necessary to provide an asm implementation here? ditto throughout

Reply:
For `_div64` specifically, yes, as compilers will generate the 64-bit encoding of `div`, which won't cause a #DE error on overflow of the 32-bit quotient, so a change of behaviour.

For everything else, generally yeah, either because it can't be represented without asm, or to ensure that the generated code behaves identically to MSVC's code, or for performance where it's relevant (like 128-bit multiplication).